src/plugins/biostruct3d_view/src/BioStruct3DGLWidget.cpp

```diff
@@ -985,8 +988,12 @@ void BioStruct3DGLWidget::sl_selectSurfaceRenderer(QAction* action) {
     update();
 }
 
-void BioStruct3DGLWidget::sl_onTaskFinished(Task* task) {
-    if (surfaceCalcTask != task || surfaceCalcTask->getState() != Task::State_Finished) {
+void BioStruct3DGLWidget::sl_onSurfaceCalcTaskFinished(Task* task) {
+    if (task->getState() != Task::State_Finished) {
+        return;
+    }
+    auto surfaceCalcTask = qobject_cast<MolecularSurfaceCalcTask*>(task);
```

Comment:
What if you have two 3D views and each of them will run 'MolecularSurfaceCalcTask' ?
Both views will receive this callback. You need to subscribe to the task itself, not to the global task scheduler instance.

Reply:
Do I understand correctly that you are suggesting that I connect the `Task::si_stateChanged()` signal to the `BioStruct3DGLWidget::sl_onTaskFinished(Task*)` slot? Based on your previous comment.